shell/platform/android/io/flutter/app/FlutterActivityDelegate.java

```diff
@@ -211,8 +244,26 @@ public void onPostResume() {
 
     @Override
     public void onDestroy() {
+        Application app = (Application) activity.getApplicationContext();
+        if (app instanceof FlutterApplication) {
+            FlutterApplication flutterApp = (FlutterApplication) app;
+            if (this.equals(flutterApp.getCurrentActivity())) {
+                Log.i(TAG, "onDestroy setting current activity to null");
+                flutterApp.setCurrentActivity(null);
+            }
+        }
         if (flutterView != null) {
-            flutterView.destroy();
+            boolean destroy = true;
+            for (ViewDestroyListener listener : viewDestroyListeners) {
+                destroy = destroy && !listener.onViewDestroy(flutterView.getFlutterNativeView());
```

Comment:
imho this would be simpler as:
```
  if (listener.onViewDestroy(flutterView.getFlutterNativeView()))
    destroy = false;
```

Reply:
Cleaned up in https://github.com/flutter/engine/pull/4357